### PR TITLE
Remove NodeDeletion Strategy

### PR DESCRIPTION
**Pull request by mshitrit**
[ECOPROJECT-1074](https://issues.redhat.com//browse/ECOPROJECT-1074) Last Phase of Removing Node Deletion Strategy, making sure it does not exist by by OLM enum validation

### CODE REVIEW

**Review by clobrano — COMMENTED** [review]
/lgtm
Giving others a chance to review as well, feel free to unhold
/hold

**Review by slintes — CHANGES_REQUESTED** [review]
please remove the webhook reference for the templates from the PROJECT file as well

File: PROJECT [hunk 1]
```diff
@@ -36,7 +33,4 @@ resources:
   kind: SelfNodeRemediationConfig
   path: github.com/medik8s/self-node-remediation/api/v1alpha1
   version: v1alpha1
-  webhooks:
```
Comment (slintes) [inline PROJECT:39]: the config still has a webhook...

**Review by slintes — CHANGES_REQUESTED** [review]

File: PROJECT (hunk 1 above)
Comment (mshitrit) [reply PROJECT:39]: Nope, it's marked in red to show it's removed.

Comment (slintes) [reply PROJECT:39]: let me rephrase: we still have a webhook for the SelfNodeRemediationConfig CRD, and this is its webhook config, which should not be deleted :)
